common/src/main/java/org/corfudb/common/metrics/micrometer/registries/RegistryProvider.java

```diff
@@ -0,0 +1,15 @@
+package org.corfudb.common.metrics.micrometer.registries;
+
+import io.micrometer.core.instrument.MeterRegistry;
+
+/**
+ * Clients who are willing to integrate their registries into Corfu metrics composite registry
+ * must implement this interface.
+ */
```

Comment:
should MeterRegistryProvider implement this interface?

Reply:
No, MeterRegistryProvider contains the CompositeRegistry. The CompositeRegistry holds the implementations of different registries (LoggingRegistry, DropwizardMeterRegistry, PrometheusMeterRegistry, etc). That way it pushes the same metrics to different backends simultaneously (this is what we want). RegistryProvider is the service provider interface, it is meant to be extended by the third parties. The parties implement this interface and provide Corfu their own configured registries. Corfu adds these registries to a CompositeRegistry and pushes all the metrics to it.